src/game/server/swarm/asw_marine.cpp

```diff
@@ -4165,6 +4180,34 @@ void CASW_Marine::Event_Killed( const CTakeDamageInfo &info )
 			}
 		}
 	}
+
+	CBaseEntity *pAttacker = info.GetAttacker();
+	CASW_Inhabitable_NPC* pAttackerNPC = dynamic_cast< CASW_Inhabitable_NPC* >( pAttacker );
+	m_bSpawnZombineOnDeath = m_bSpawnZombineOnDeath ||
+							( pAttackerNPC && pAttackerNPC->m_bSpawnZombineOnMarineKill ) ||
+							( pAttackerNPC && pAttackerNPC->Classify() != CLASS_ASW_MARINE && rd_marine_spawn_zombine_on_death_chance.GetFloat() > RandomFloat() );
+
+	if ( m_bSpawnZombineOnDeath )
+	{
+		CNPC_Zombine* pZombine = dynamic_cast< CNPC_Zombine* >( CreateEntityByName( "npc_zombine" ) );
+
+		if ( !pZombine )
+		{
+			Warning( "Unable to create npc_zombine from dying marine.\n" );
+		}
+		else
+		{
+			pZombine->SetAbsOrigin( GetAbsOrigin() + Vector( 0.0, 0.0, 10.0 ) ); // spawns in the ground, raise height by 10
+			pZombine->SetAbsAngles( GetAbsAngles() );
+
+			pZombine->Spawn();
+
+			pZombine->SetModel( "models/zombie/fallen_marine.mdl" );
```

Comment:
Either move the precache of this model to CASW_Marine or move the SetModel call to a method on zombine.